bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,242 @@
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $NF}') = "nvidia" ]; then
+    echo true
+else
+    echo false
+fi
+'''], condition = { env_not_set = ["GPU"] } }
+
+[tasks.mgmt-cluster-kubeconfig-base64.env]
+MGMT_CLUSTER_KUBECONFIG_BASE64 = {script=['''
+if [ -n "${MGMT_CLUSTER_KUBECONFIG_PATH}" ]; then
+    cat ${MGMT_CLUSTER_KUBECONFIG_PATH} | base64
+fi
+''']}
+
+[tasks.instance-types]
+dependencies = ["gpu"]
+
+[tasks.instance-types.env]
+INSTANCE_TYPES = {script=['''
+if [ -z ${INSTANCE_TYPES} ]; then
+    if [ ${ARCHITECTURE} = "x86_64" ]; then
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g4dn.xlarge\"]"
+        else
+            echo "[\"m5.large\"]"
+        fi
+    else
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g5g.xlarge\"]"
+        else
+            echo "[\"m6g.large\"]"
+        fi
+    fi
+else
+    echo ${INSTANCE_TYPES}
+fi
+''']}
+
+[tasks.set-env]
+dependencies = ["set-agent-images", "cluster-name", "gpu", "metadata-url", "ova-name", "bottlerocket-ami-id", "mgmt-cluster-kubeconfig-base64", "instance-types"]
+
+[tasks.create-inner]
+dependencies = ["set-env"]
+script_runner = "bash"
+script = '''
+eval "cat > ${OUTPUT_FILE} << EOF
+$(< ${CLUSTER_TYPE}/${TEST_TYPE}.yaml)
+EOF
+" 2> /dev/null
+
+echo "${OUTPUT_FILE}"
+'''
+
+[tasks.create-inner.env]
+OUTPUT_FILE = {script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    echo "output/${CLUSTER_NAME}-migration.yaml"
+elif [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "workload" ]; then
+    echo "output/${CLUSTER_NAME}-workload.yaml"
+else
+    echo "output/${CLUSTER_NAME}.yaml"
+fi
+''']}
+
+[tasks.create]
+script = '''
+#!@duckscript
+set_env TEST_TYPE ${1}
+cm_run_task ${1}
+cm_run_task create-inner
+'''
+
+[tasks.install-controller]
+dependencies = ["set-agent-images"]
+script = "cli install --controller-uri ${CONTROLLER_IMAGE_URI}"
+
+[tasks.add-aws-secret]
+dependencies = ["install-controller"]
+condition = { env_contains = { "CLUSTER_TYPE" = "kind", "VARIANT" = "aws" } }
+script = '''
+ACCESS_KEY_ID=$(aws configure get aws_access_key_id)
+SECRET_ACCESS_KEY=$(aws configure get aws_secret_access_key)
+cli add-secret map  \
+ --name "aws-creds" \
+ "ACCESS_KEY_ID=${ACCESS_KEY_ID}" \
+ "SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY}"
+'''
+
+[tasks.add-vsphere-secret]
+dependencies = ["install-controller"]
+condition = { env_contains = { "VARIANT" = "vmware" } }
+script = '''
+cli add-secret map  \
+ --name "vsphere-creds" \
+ "username=${GOVC_USERNAME}" \
+ "password=${GOVC_PASSWORD}"
+'''
+
+[tasks.run-inner]
+dependencies = ["install-controller", "add-aws-secret", "add-vsphere-secret"]
+script = "cli run file ${OUTPUT_FILE}"
+
+[tasks.run]
+dependencies = ["create"]
+script = '''
+#!@duckscript
+cm_run_task run-inner
+'''
+
+[tasks.ecs-migration-test.env]
+VARIANT = "aws-ecs-1"
+
+[tasks.ecs-test.env]
+VARIANT = "aws-ecs-1"
+
+[tasks.ecs-workload-test.env]
+VARIANT = "aws-ecs-1-nvidia"
```

Comment:
Maybe you could set things like GPU and instance type here instead of the `awk`robatics.

Reply:
I wanted the effort needed to add a sample to be the absolute minimum, which is why only the variant is set here. I think I prefer it this way.

Reply:
The effort to add a sample for a variant that ends with `-xyz` and has a workload test is higher this way.

Reply:
@ecpullen wanted a separate target for each variable and I'm inclined to agree that it looks cleaner than putting several variables inside the `test` target. Leaving it the way it is now also makes it easier for others to add their own sample files and targets.